test/Dialect/SSP/roundtrip.mlir

```diff
@@ -142,64 +166,80 @@ ssp.instance @mco_outgoing_delays of "ChainingProblem" {
+// CHECK:     %[[op_4:.*]] = operation<@fastAdd>() [rsrc<@fastAdder>, t<1>]
 // CHECK:     %[[op_5:.*]] = operation<@_0>(%[[op_0]], %[[op_1]], %[[op_2]], %[[op_3]], %[[op_4]]) [t<10>]
 // CHECK:     operation<@_1>() [t<10>]
 // CHECK:   }
 // CHECK: }
 ssp.instance @multiple_oprs of "SharedOperatorsProblem" {
   library {
-    operator_type @slowAdd [latency<3>, limit<2>]
-    operator_type @fastAdd [latency<1>, limit<1>]
+    operator_type @slowAdd [latency<3>]
+    operator_type @fastAdd [latency<1>]
     operator_type @_0 [latency<0>]
     operator_type @_1 [latency<1>]
   }
+  resource {
+    resource_type @slowAdder [limit<1>]
+    resource_type @fastAdder [limit<1>]
+  }
   graph {
-    %0 = operation<@slowAdd>() [t<0>]
-    %1 = operation<@slowAdd>() [t<1>]
-    %2 = operation<@fastAdd>() [t<0>]
-    %3 = operation<@slowAdd>() [t<1>]
-    %4 = operation<@fastAdd>() [t<1>]
+    %0 = operation<@slowAdd> uses<@slowAdder>() [t<0>]
+    %1 = operation<@slowAdd> uses<@slowAdder>() [t<1>]
+    %2 = operation<@fastAdd> uses<@fastAdder>() [t<0>]
+    %3 = operation<@slowAdd> uses<@slowAdder>() [t<1>]
+    %4 = operation<@fastAdd> uses<@fastAdder>() [t<1>]
     %5 = operation<@_0>(%0, %1, %2, %3, %4) [t<10>]
     operation<@_1>() [t<10>]
   }
 }
 
 // CHECK: ssp.instance @canis14_fig2 of "ModuloProblem" [II<3>] {
 // CHECK:   library {
-// CHECK:     operator_type @MemPort [latency<1>, limit<1>]
+// CHECK:     operator_type @MemAccess [latency<1>]
 // CHECK:     operator_type @Add [latency<1>]
 // CHECK:     operator_type @Implicit [latency<0>]
 // CHECK:   }
+// CHECK:  resource {
+// CHECK:    resource_type @ReadPort [limit<1>]
+// CHECK:    resource_type @WritePort [limit<1>]
+// CHECK:  }
 // CHECK:   graph {
-// CHECK:     %[[op_0:.*]] = operation<@MemPort> @load_A(@store_A [dist<1>]) [t<2>]
-// CHECK:     %[[op_1:.*]] = operation<@MemPort> @load_B() [t<0>]
+// CHECK:     %[[op_0:.*]] = operation<@MemAccess> @load_A(@store_A [dist<1>]) [rsrc<@ReadPort>, t<2>]
+// CHECK:     %[[op_1:.*]] = operation<@MemAccess> @load_B() [rsrc<@ReadPort>, t<0>]
 // CHECK:     %[[op_2:.*]] = operation<@Add> @add(%[[op_0]], %[[op_1]]) [t<3>]
-// CHECK:     operation<@MemPort> @store_A(%[[op_2]]) [t<4>]
+// CHECK:     operation<@MemAccess> @store_A(%[[op_2]]) [rsrc<@WritePort>, t<4>]
 // CHECK:     operation<@Implicit> @last(@store_A) [t<5>]
 // CHECK:   }
 // CHECK: }
 ssp.instance @canis14_fig2 of "ModuloProblem" [II<3>] {
   library {
-    operator_type @MemPort [latency<1>, limit<1>]
+    operator_type @MemAccess [latency<1>]
     operator_type @Add [latency<1>]
     operator_type @Implicit [latency<0>]
   }
+  resource {
+    resource_type @ReadPort [limit<1>]
+    resource_type @WritePort [limit<1>]
+  }
   graph {
-    %0 = operation<@MemPort> @load_A(@store_A [dist<1>]) [t<2>]
-    %1 = operation<@MemPort> @load_B() [t<0>]
+    %0 = operation<@MemAccess> uses<@ReadPort>@load_A(@store_A [dist<1>]) [t<2>]
```

Comment:
We should change uses to take an array of resources and for schedulers that do not support multiple resource types yet error out.